homeassistant/components/depict.py

```diff
@@ -0,0 +1,100 @@
+"""
+Support for controlling Depict digital art frames.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/depict/
+"""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    CONF_HOST,
+    CONF_NAME,
+    EVENT_HOMEASSISTANT_STOP
+)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+REQUIREMENTS = ['depict-control==1.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_VALUE = 'value'
+DATA_DEPICT = 'depict'
+DOMAIN = 'depict'
+SIGNAL_SET_CONTRAST = 'depict.set_contrast'
+
+AUTODETECT_SCHEMA = vol.Schema({})
+
+FRAME_SCHEMA = vol.Schema({
+    vol.Required(CONF_NAME): cv.string,
+    vol.Required(CONF_HOST): cv.string,
+})
+
+FRAMES_SCHEMA = vol.Schema([FRAME_SCHEMA])
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Any(AUTODETECT_SCHEMA, FRAMES_SCHEMA),
+}, extra=vol.ALLOW_EXTRA)
+
+SERVICE_SET_CONTRAST_SCHEMA = vol.Schema({
+    ATTR_ENTITY_ID: cv.entity_ids,
+    ATTR_VALUE: cv.byte,
+})
+
+
+async def async_setup(hass, config):
+    """Set up the depict component."""
+    from depict_control import Frame
+    frames = hass.data.setdefault(DATA_DEPICT, {})
+    frame_configs = config.get(DOMAIN)
+    session = async_get_clientsession(hass)
+
+    async def add_frame(host, name=None):
+        """Add platforms for a single frame with the given hostname."""
+        frame = await Frame.connect(session, host)
+        if name is None:
+            name = frame.name
+        frames[name] = frame
```

Comment:
Yes. In the auto-detect case I'm not sure what else to do, though.

Reply:
Well maybe could generate a random key (random int for example) instead of using the name as the dictionary key. Then pass that key along in `async_load_platform`

Reply:
If I've got two frames with the same name, I've set myself up for a confusing experience in the Depict app. (So it should be rare.) I think letting Hass automatically add a number (e.g. `_2`) to the name in that case is fine, and preferable to decorating all discovered names.